src/test/java/AlertServiceTest.java

```diff
@@ -0,0 +1,123 @@
+
+@ExtendWith(MockitoExtension.class)
+public class AlertServiceTest {
+
+    @Mock
+    private HttpClient mockClient;
+    @Mock
+    private GamesFetcher mockGamesFetcher;
+
+    @InjectMocks
+    private AlertService alertService;
+
+    @BeforeEach
+    public void setUp() {
+        alertService = new AlertService();
+        alertService.setHttpClient(mockClient);
+    }
+
+    @Test
+    public void testCreateOrUpdateAlertShouldReturnCorrectInput() throws IOException, InterruptedException {
+
+        HttpResponse<String> mockResponse = Mockito.mock(HttpResponse.class);
+
+        when(mockResponse.body()).thenReturn("true");
+        when(mockClient.send(any(HttpRequest.class),
+                any(HttpResponse.BodyHandlers.ofString().getClass()))).thenReturn(mockResponse);
+
+        boolean result = alertService.createOrUpdateAlert("test@example.com", 1, 9.99);
+
+        assertTrue(result);
+    }
+
+    @Test
+    public void testGetAlertsForEmailShouldReturnCorrectResponse() throws IOException, InterruptedException {
+
+        HttpResponse<String> mockResponse = Mockito.mock(HttpResponse.class);
+
+        when(mockResponse.body()).thenReturn("Example link send");
+        when(mockClient.send(any(HttpRequest.class),
+                any(HttpResponse.BodyHandlers.ofString().getClass()))).thenReturn(mockResponse);
+
+        String result = alertService.getAlertsForEmail("test@example.com");
+
+        assertEquals("Example link send", result);
+    }
+
+    @Test
+    public void testCreateOrUpdateAlertWithGameTitleShouldReturnCorrectInput() throws IOException, InterruptedException {
+
+        GamePojo mockGamePojo = new GamePojo();
+        mockGamePojo.setTitle("Test title");
+
+        when(mockGamesFetcher.getGameContainingKeyword(anyString())).thenReturn(List.of(mockGamePojo));
+
+        HttpResponse<String> mockResponse = Mockito.mock(HttpResponse.class);
+
+        when(mockResponse.body()).thenReturn("true");
+        when(mockClient.send(any(HttpRequest.class),
+                any(HttpResponse.BodyHandlers.ofString().getClass()))).thenReturn(mockResponse);
+
+        alertService.setGamesFetcher(mockGamesFetcher);
+
+        boolean result = alertService.createOrUpdateAlertWithGameTitle("Test title", "test@example.com", 4.45);
+
+        assertTrue(result);
+    }
+
+    @Test
+    public void testDeleteAlertShouldReturnTrueOnSuccess() throws IOException, InterruptedException {
+
+        HttpResponse<String> mockResponse = Mockito.mock(HttpResponse.class);
+
+        when(mockResponse.body()).thenReturn("true");
+        when(mockClient.send(any(HttpRequest.class),
+                any(HttpResponse.BodyHandlers.ofString().getClass()))).thenReturn(mockResponse);
+
+
+        boolean result = alertService.deleteAlert("test@example.com", 1);
+
+        assertTrue(result);
+    }
+
+    @Test
+    public void testCreateOrUpdateAlertWithGameTitleShouldReturnFalseWhenGameNotFound() throws IOException, InterruptedException {
```

Comment:
Could you also add a case where the game title is null?

Reply:
Added tests to handle null, also added if checks in my method to deal with it.